docs/references/errors.rst

```diff
@@ -151,6 +151,43 @@ Returns:
     "code": "PT402"
   }
 
+.. _raise_headers:
+
+Add HTTP Headers with RAISE error
+---------------------------------
+
+You can add custom HTTP status and  headers to the response by raising a ``PGRST`` SQLSTATE error. You can achieve this by adding the ``code``, ``message``, ``detail`` and ``hint`` in the postgresql error message field as a JSON object. Here, the ``details`` and ``hint`` are optional. Similarly, the ``status`` and ``headers`` must be added to the SQL error detail field as a JSON object. For instance:
+
+.. code-block:: sql
+
+  RAISE sqlstate 'PGRST' USING
+      message = '{"code":"123","message":"Payment Required","details":"Quota exceeded","hint":"Upgrade your plan"}',
+      detail = '{"status":402,"headers":{"X-Powered-By":"PHP/5.4.0"}}';
```

Comment:
Why PHP? :smile:

How about one of https://www.pingdom.com/blog/fun-and-unusual-http-response-headers/ :stuck_out_tongue_closed_eyes: 

```suggestion
      detail = '{"status":402,"headers":{"X-Powered-By":"Nerd Rage"}}';
```